src/Zoo.cpp

```diff
@@ -0,0 +1,262 @@
+	default:
+		break;
+	}
+}
+
+void Zoo::del(int index)
+{
+	if (index < 0 || index >= animals.size())
+		throw std::out_of_range("Index out of range\n");
+	animals.erase(animals.begin() + index);
+}
+
+void Zoo::delAll(AnimalType type)
+{
+	char initial;
+	switch (type)
+	{
+	case CLOWNFISH:
+		initial = 'C';
+		break;
+	case DOG:
+		initial = 'D';
+		break;
+	case GOOSE:
+		initial = 'G';
+		break;
+	case LION:
+		initial = 'L';
+		break;
+	case MONKEY:
+		initial = 'M';
+		break;
+	case OWL:
+		initial = 'O';
+		break;
+	case SHARK:
+		initial = 'S';
+		break;
+	default:
+		initial = 'a';
+		break;
+	}
+	animals.erase(
+		std::remove_if(animals.begin(), animals.end(),
+			[initial](const unique_ptr<Animal>& a) {  return a->getInitial() == initial; }),
+		animals.end());
+}
+
+void Zoo::help() const
+{
+	cout << "Help-menu:\n";
+	cout << "'stop': Make animal at index (which should be received in the next input) to stop\n";
+	cout << "'move': Make animal at index (which should be received in the next input) to move\n";
+	cout << "'create': Creates new animal with type and (that receives from the user in the next inputs)\n";
+	cout << "'del': Delete animal at index (which should be received in the next input)\n";
+	cout << "'delAll': Removes all animal of specific type (receives the type in the next input)\n";
+	cout << "'.': Moving to the next turn without change\n";
+	cout << "'exit': Exit the program\n";
+	cout << "'help': Print the menu of all available commands\n";
+	cout << endl;
+}
+
+void Zoo::printMap() const
+{
+	cout << "Zoo-Map:\n";
+	for (int r = 0; r < ROWS; ++r)
+	{
+		for (int c = 0; c < COLUMNS; c++)
+		{
+			Location loc(r, c);
+			bool is_full = false;
+			for (auto& animal : animals)
+			{
+				if (animal->getLocation() == loc)
+				{
+					is_full = true;
+					cout << animal->getInitial() << " ";
+					break;
+				}
+			}
+			if (!is_full)
+				cout << "- ";
+		}
+		cout << endl;
+	}
+	cout << endl;
+}
+
+void Zoo::printList() const
+{
+	cout << "Zoo-List:\n";
+	int index = 0;
+	for (auto& animal : animals)
+	{
+		cout << index << ". ";
+		animal->printDetails();
+		cout << endl;
+		++index;
+	}
+	cout << endl;
+}
+
+Zoo::AnimalType Zoo::toAnimalType(const std::string& str) const
+{
+
+	static const std::map<std::string, AnimalType> animalMap =
+	{
+		{"clownfish", CLOWNFISH},
+		{"dog", DOG},
+		{"goose", GOOSE},
+		{"lion", LION},
+		{"monkey", MONKEY},
+		{"owl", OWL},
+		{"shark", SHARK}
+	};
+	auto it = animalMap.find(str);
+	if (it == animalMap.end())
+		throw std::invalid_argument("Invalid animal type string: " + str);
+	return it->second;
+}
+
+void Zoo::toLower(string &s) const
+{
+	std::transform(s.begin(), s.end(), s.begin(), ::tolower);
+}
+
+Zoo::Zoo()
+{
+	create(LION, "Simba");
+	create(LION, "Mufasa");
+	create(MONKEY, "Rafiki");
+	create(GOOSE, "Akka");
+	create(GOOSE, "Morten");
+	create(OWL, "Hedwig");
+	create(DOG, "Toto");
+	create(SHARK, "Jaws");
+	create(CLOWNFISH, "Nemo");
+}
+
+void Zoo::run()
+{
```

Comment:
function run is handling multiple responsibilities, such as displaying the zoo map,
printing animal details, processing user commands. you can declare the command map in the c-tor